include/ode_solvers/euler.hpp

```diff
@@ -11,7 +11,7 @@
 #ifndef EULER_HPP
 #define EULER_HPP
 
-template <typename Point, typename NT, class Polytope, class func=std::function <Point(std::vector<Point>, NT)>>
+template <typename Point, typename NT, class Polytope, class func=std::function <Point(std::vector<Point>&, NT&)>>
 class EulerODESolver {
```

Comment:
As a general principle, I do not like the use of pointers. In this context, I think that we should consider another implementation of the bounds. 
We expect that Polytope is a big object and thus we should rely on move semantics and not on pointers to handle it.  
Also, I think that the size of bounds should be known in compile time.